architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,446 @@
+
+He first sets up the Radius Helm chart as a dependency in the cluster, and then runs the `rad bicep generate bicepdeployment env.bicep --parameters env.bicepparam` command on his machine to create an BicepDeployment resource that contains the template and parameters specified. He then commits the BicepDeployment resource to the Git repository.
+
+When a developer creates a new development cluster using the configuration specified in Jon's GitHub repository with a GitOps tool, the BicepDeployment resource is applied on the cluster, and the Radius environment defined in the `env.bicep` Bicep manifest is deployed. The developer can then use the environment to deploy and test their applications using Radius.
+
+#### Bran can use GitOps to deploy and manage a Radius application in production
+
+Bran is a developer who uses GitOps to automate the deployment of Radius applications to production Kubernetes clusters. His infrastructure team has set up a Radius installation and environment on the production cluster, and Bran wants to deploy his application using Radius. He has authored a Bicep manifest that defines the resources for his application and a Bicep parameters file that specifies the parameters for the manifest. He runs the `rad bicep generate bicepdeployment app.bicep --parameters app.bicepparam` command on his machine to create an BicepDeployment resource that contains the template and parameters specified. He then commits the BicepDeployment resource to the Git repository.
+
+When the GitOps tool watches the repository and applies the BicepDeployment resource to the production cluster, the Radius resources defined in the `app.bicep` Bicep manifest are deployed.
+
+#### Sam can use GitOps to patch a Radius application in production
+
+Sam is an SRE who is responsible for maintaining a Radius application in production. He notices that the application is experiencing performance issues and the number of application container replicas needs to be increased. He updates the Bicep parameters file for the application to specify the new number of replicas and runs the `rad bicep generate bicepdeployment app.bicep --parameters app.bicepparam` command on his machine to create an BicepDeployment resource that contains the updated parameters. He then commits the BicepDeployment resource to the Git repository.
+
+When the GitOps tool watches the repository and applies the BicepDeployment resource to the production cluster, the Radius resources defined in the `app.bicep` Bicep manifest are updated with the new parameters. The application container replicas are increased, and the performance issues are resolved.
+
```

Comment:
Does it make sense to scope these scenarios to `BicepDeployment`? These are the same end-to-end scenarios we have for Gitops overall right?

Reply:
let's move them out